subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -554,15 +547,15 @@ int lwm2m_init_message(struct net_context *net_ctx, struct zoap_packet *zpkt,
 	return 0;
 }
 
-struct zoap_pending *lwm2m_init_message_pending(struct zoap_packet *zpkt,
-						struct sockaddr *addr,
-						struct zoap_pending *zpendings,
-						int num_zpendings)
+struct zoap_pending *lwm2m_init_message_pending(struct lwm2m_ctx *client_ctx,
+						struct zoap_packet *zpkt,
+						struct sockaddr *addr)
```

Comment:
I suppose "addr" can be removed. Peer sockaddr can be obtained from client_ctx->net_app_ctx->default_ctx->remote

Reply:
Eventually the addr here will be resolved from the security objects (see page 121 of the LwM2M spec example URL: coaps://bootstrap.example.com).   If the 0/0/0 resource is a bootstrap server it will then point us to another server which would need to be resolved, etc.

Reply:
@rbtchc Your original comment here now applies.  Going forward the rd_client will start engine with peer_addr information on a per server basis.  So net_app_ctx.default_ctx->remote will be the valid destination address as far as the engine is concerned.

I will add a commit to V3 addressing this.

Reply:
I also retract my earlier agreement.  As with lwm2m_udp_sendto(), this function is used by the observer notify logic.  And there shouldn't be an inherent limitation as to who can subscribe to notifications except where access controls (to be added later) allow.